unity/Assets/URDF-Loader/URDFParser.cs

```diff
@@ -64,55 +89,114 @@ public class URDFParser : MonoBehaviour {
                         }
                         visRot = URDFToUnityRot(visRot);
-                        
-                        // load the STL file if possible
-                        try {
+
+                        try
+                        {
+                            // try to load primitives if there is no mesh
+                            if (meshNode == null)
+                            {
+                                XmlNode primitiveNode = GetXmlNodeChildByName(geomNode, "box") ??
+                                                        GetXmlNodeChildByName(geomNode, "sphere") ??
+                                                        GetXmlNodeChildByName(geomNode, "cylinder");
+                                if (primitiveNode == null)
+                                    continue;
+                                GameObject go = null;
+                                switch (primitiveNode.Name)
+                                {
+                                    case "box":
+                                        go = GameObject.CreatePrimitive(PrimitiveType.Cube);
+                                        go.transform.localScale =
+                                            URDFToUnityPos(TupleToVector3(primitiveNode.Attributes[0].Value));
+                                        break;
+                                    case "sphere":
+                                        go = GameObject.CreatePrimitive(PrimitiveType.Sphere);
+                                        go.transform.localScale =
+                                            Vector3.one * (float.Parse(primitiveNode.Attributes[0].Value) * 2);
+                                        break;
+                                    case "cylinder":
+                                        go = new GameObject();
+                                        var cPrimitive = GameObject.CreatePrimitive(PrimitiveType.Cylinder);
+                                        cPrimitive.transform.parent = go.transform;
+
+                                        var length = float.Parse(primitiveNode.Attributes[0].Value);
+                                        var radius = float.Parse(primitiveNode.Attributes[1].Value);
+                                        go.transform.localScale = new Vector3(radius * 2, length, radius * 2);
+                                        break;
+                                }
+                                Renderer r = go.GetComponent<Renderer>() ?? go.GetComponentInChildren<Renderer>();
+                                go.transform.parent = urdfLink.transform;
+                                go.transform.localPosition = visPos;
+                                go.transform.localRotation = Quaternion.Euler(visRot);
+
+                                go.name = urdfLink.name + " geometry " + primitiveNode.Name;
+
+                                if (!r)
+                                    continue;
```

Comment:
Can we do this portion without continue statements? It makes it a little harder to follow